packages/react/src/components/OptionsMenu/OptionsMenuList.tsx

```diff
@@ -9,58 +9,42 @@ const [up, down, tab, enter, space, esc] = [38, 40, 9, 13, 32, 27];
 export interface OptionsMenuListProps
   extends Omit<OptionsMenuProps, 'trigger'> {
   className?: string;
+  triggerRef: React.RefObject<HTMLButtonElement>;
 }
```

Comment:
Unfortunately, this appears to be a breaking change. Does this change still need to be made even when we're checking the event in the capture phase?

Reply:
This change is needed because the trigger button is outside of the `ClickOutsideListener` component which cause the menu to not be able to close after it was opened.

Reply:
I pulled down your PR, and I have a few thoughts about this.

One way we could handle this in a non-breaking manner, is to check for `event.defaultPrevented` here: 

https://github.com/dequelabs/cauldron/blob/9df13e6645fce1903a742900c9b0db304a334546/packages/react/src/components/OptionsMenu/OptionsMenu.tsx#L48-L50

Within the click outside callback we could then set `event.preventDefault` if the menu list is currently shown. This does come with the downside of it would prevent immediate clicks on elements outside of the menu while the list is open so not the best solution, but also non-breaking.

Aside from that, we need to consider these components could also be constructed piecemeal so we need to accommodate for that scenario as well. Instead of explicitly trying to set the ref on a component (and not all components correctly implement the `ref` property) we might want to consider capturing the element that triggered the open state:

```ts
const triggerRef = useRef<HTMLElement>()
const toggleMenu = (event) => {
  if (!show) {
    triggerRef.current = document.activeElement
  } else {
    triggerRef.current = null
  }
  setShow(!show);
};
```

OptionsMenuList could _optionally_ accept the ref (to be non breaking):

```ts
export interface OptionsMenuListProps
  extends Omit<OptionsMenuProps, 'trigger'> {
  className?: string;
  triggerRef?: React.RefObject<HTMLElement>
}

...

const handleClickOutside = (e: MouseEvent | TouchEvent) => {
  if (triggerRef && triggerRef.current) {
    // use existing checks
  }
}
```

IMO it makes it slightly more robust without there needing to be a breaking change. Thoughts?

Reply:
In the long run, we're on the path to create a brand new component that will replace this one and deprecate usage of this component. It has a lot odd patterns that do not support some of the mechanisms we're trying to achieve and any changes we would make would need to be breaking. A brand new component to move away from some of these patterns without breaking immediate usage is the path of least resistance.

Reply:
> I pulled down your PR, and I have a few thoughts about this.
> 
> One way we could handle this in a non-breaking manner, is to check for `event.defaultPrevented` here:
> 
> https://github.com/dequelabs/cauldron/blob/9df13e6645fce1903a742900c9b0db304a334546/packages/react/src/components/OptionsMenu/OptionsMenu.tsx#L48-L50
> 
> Within the click outside callback we could then set `event.preventDefault` if the menu list is currently shown. This does come with the downside of it would prevent immediate clicks on elements outside of the menu while the list is open so not the best solution, but also non-breaking.

I tried out what you said above and see that it works for the `OptionsMenu` component but it doesn't fix the issue where the `OptionsMenuList` is within the `TopBarMenu` in the `TopBar` example.

> Aside from that, we need to consider these components could also be constructed piecemeal so we need to accommodate for that scenario as well. Instead of explicitly trying to set the ref on a component (and not all components correctly implement the `ref` property) we might want to consider capturing the element that triggered the open state:
> 
> ```ts
> const triggerRef = useRef<HTMLElement>()
> const toggleMenu = (event) => {
>   if (!show) {
>     triggerRef.current = document.activeElement
>   } else {
>     triggerRef.current = null
>   }
>   setShow(!show);
> };
> ```
> 
> OptionsMenuList could _optionally_ accept the ref (to be non breaking):
> 
> ```ts
> export interface OptionsMenuListProps
>   extends Omit<OptionsMenuProps, 'trigger'> {
>   className?: string;
>   triggerRef?: React.RefObject<HTMLElement>
> }
> 
> ...
> 
> const handleClickOutside = (e: MouseEvent | TouchEvent) => {
>   if (triggerRef && triggerRef.current) {
>     // use existing checks
>   }
> }
> ```
> 
> IMO it makes it slightly more robust without there needing to be a breaking change. Thoughts?

If you look at my changes for the `TopBar` docs, I set the `menuItemRef` and then pass that to the `OptionsMenuList` which would accept it optionally. Even if the triggerRef prop is optional, the OptionsMenuList still wouldn't open unless users set up the `menuItemRef`.

https://github.com/dequelabs/cauldron/blob/555e12f0f032b584fca04d427a54446c79c12409/docs/pages/components/TopBar.mdx?plain=1#L65-L85

Reply:
Updated here to not use a ref prop https://github.com/dequelabs/cauldron/pull/1680/commits/aa0642d9087648fb1c4636e354a19187067aec62